packages/loopring_v3/contracts/impl/Exchange.sol

```diff
@@ -49,18 +50,43 @@ contract Exchange is IExchange, Claimable, ReentrancyGuard
     using ExchangeWithdrawals   for ExchangeData.State;
 
     ExchangeData.State private state;
-    // -- Constructor --
-    constructor(
-        uint    _id,
-        address payable _loopringAddress,
+
+    modifier onlyOperator()
+    {
+        require(msg.sender == state.operator, "UNAUTHORIZED");
+        _;
+    }
+
+    modifier onlyWhenUninitialized()
+    {
+        require(owner == address(0) && state.id == 0, "INITIALIZED");
+        _;
+    }
+
+    /// @dev The constructor must do NOTHING to support proxy.
+    constructor() public {}
+
+    /// @dev The default function will do Ether depoeits.
+    function() external payable
+    {
+        uint96 amount = uint96(msg.value);
+        require(uint(amount) == msg.value, "TOO_LARGE");
+        state.depositTo(msg.sender, address(0), amount, 0);
```

Comment:
Currently the `amount` in the deposit is the actual amount the user wants to deposit, without the fee (to make the behavior the same with ERC20 tokens + to guard users against sudden increased deposit fees). So this will only work if the deposit fee of the exchange is 0. Subtracting the deposit fee here is possible (so that `amount - depositFee` is deposited) but is a bit dangerous for users for the reason given above.

Reply:
do you suggest we should disable this default function?

Reply:
I should have fixed the issue, but maybe we should simply disable the default function.

Reply:
I also think it's best to disable the default function here (by not defining any).

Reply:
OK, will do.